File: tests/unit/X2Mcp.Language.Go.Tests/PublicFunctions.go
package fixtures

func Add(a, b int) int {
	return a + b
}

func Greet(name string) string {
	return "Hello, " + name
}

func LogMessage(msg string) {
	println(msg)
}

func Validate(input string) error {
	return nil
}

func Divide(a, b float64) (float64, error) {
	return a / b, nil
}

File: tests/unit/X2Mcp.Language.Rust.Tests/Fixtures/public_functions.rs
pub fn add(a: i32, b: i32) -> i32 {
    a + b
}

pub fn greet(name: String) -> String {
    format!("Hello, {}!", name)
}

pub fn log_message(msg: String) {
    println!("{}", msg);
}

pub fn validate(input: String) -> Result<(), String> {
    Ok(())
}

pub fn divide(a: f64, b: f64) -> Result<f64, String> {
    if b == 0.0 {
        return Err("division by zero".to_string());
    }
    Ok(a / b)
}

pub fn format_value(value: f64, format: Option<String>) -> String {
    format!("{:?} {:?}", value, format)
}
